prow/cmd/mkbuild-cluster/main.go

```diff
@@ -234,12 +238,37 @@ func do(o options) error {
 
 	// Try to use this entry
 	if !o.skipCheck {
-		c, err := kube.NewClient(&newCluster, "kube-system")
+		cfg := &clientcmdapi.Config{
+			Clusters: map[string]*clientcmdapi.Cluster{
+				o.alias: {
+					Server:                   newCluster.Endpoint,
+					CertificateAuthorityData: newCluster.ClusterCACertificate,
+				},
+			},
+			AuthInfos: map[string]*clientcmdapi.AuthInfo{
+				o.alias: {
+					ClientCertificateData: newCluster.ClientCertificate,
+					ClientKeyData:         newCluster.ClientKey,
+				},
+			},
+			Contexts: map[string]*clientcmdapi.Context{
+				o.alias: {
+					Cluster:  o.alias,
+					AuthInfo: o.alias,
+				},
+			},
+		}
+
+		contextCfg, err := clientcmd.NewNonInteractiveClientConfig(*cfg, o.alias, &clientcmd.ConfigOverrides{}, nil).ClientConfig()
+		if err != nil {
+			return fmt.Errorf("create %s client: %v", o.alias, err)
+		}
+		client, err := kubernetes.NewForConfig(contextCfg)
 		if err != nil {
-			return err
+			return fmt.Errorf("could not create kubernetes client: %v", err)
 		}
-		if _, err = c.ListPods("k8s-app=kube-dns"); err != nil {
```

Comment:
The point of this check is to ensure the user has a ClusterRoleBinding for manipulating pods